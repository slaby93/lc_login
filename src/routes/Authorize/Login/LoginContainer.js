import React from 'react';
import { connect } from 'react-redux';
import Login from './Login';
import * as USER_ACTIONS from './../../../ducks/user';

export const LoginContainer = (props) => {
  return (<Login {...props} />);
};

export function mapStateToProps(state) {
  return {
    isLogging: state.getIn(['user', 'isLogging']),
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    onSubmit: ({ email, password, rememberMe }) => {
      dispatch(USER_ACTIONS.USERS_LOGIN_REQUEST({
        email,
        password,
        rememberMe,
      }));
    },
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginContainer);
