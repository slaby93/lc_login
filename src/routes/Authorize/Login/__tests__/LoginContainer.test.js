import React from 'react';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';
import { LoginContainer, mapStateToProps, mapDispatchToProps } from './../LoginContainer';

jest.mock('./../Login');

describe('<LoginContainer />', () => {
  test('basic render', () => {
    const mountedComponent = shallow(<LoginContainer />);
    expect(mountedComponent).toMatchSnapshot();
  });

  test('mapStateToProps', () => {
    expect(mapStateToProps(fromJS({
      user: {
        isLoggin: true,
      },
    }))).toMatchSnapshot();
    expect(mapStateToProps(fromJS({
      user: {
        isLoggin: false,
      },
    }))).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    const dispatch = jest.fn();

    test('result shape', () => {
      dispatch.mockReset();
      const result = mapDispatchToProps(dispatch);
      expect(result).toMatchSnapshot();
    });

    test('onSubmit', () => {
      dispatch.mockReset();
      const result = mapDispatchToProps(dispatch);
      const data = { email: 'test1', password: 'test2', rememberMe: 'test3' };
      result.onSubmit(data);
      expect(dispatch).toBeCalledWith({ type: 'USER/LOGIN/REQUEST', payload: { ...data } });
    });
  });
});
