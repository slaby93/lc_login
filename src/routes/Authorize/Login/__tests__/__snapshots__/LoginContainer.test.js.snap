// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginContainer /> basic render 1`] = `<Login__StyledComponent />`;

exports[`<LoginContainer /> mapDispatchToProps result shape 1`] = `
Object {
  "onSubmit": [Function],
}
`;

exports[`<LoginContainer /> mapStateToProps 1`] = `
Object {
  "isLogging": undefined,
}
`;

exports[`<LoginContainer /> mapStateToProps 2`] = `
Object {
  "isLogging": undefined,
}
`;
